### ISSUE TITLE
Able to set AUTO_INCREMENT to a value larger than maximum value of int

### ISSUE THREAD

**Issue by fubinzh** [pingcap/tidb#56034]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
1. create a table with auto increment, and set AUTO_INCREMENT=2147483648, while max value for int is 2147483647 (https://docs.pingcap.com/tidb/stable/data-type-numeric#integer-type)
2. insert into the value using auto increment
```
mysql> create table t1 (a int auto_increment) AUTO_INCREMENT=2147483648;
Query OK, 0 rows affected (0.07 sec)

mysql> insert into t1 values ();
ERROR 1690 (22003): constant 2147483648 overflows int

```

### 2. What did you expect to see? (Required)
Create table should fail if the AUTO_INCREMENT set is not valid

### 3. What did you see instead (Required)
Create table is OK, but later insert fails.

### 4. What is your TiDB version? (Required)
v8.4.0-alpha
Edition: Community
Git Commit Hash: f9c4773ba34854cec93dec3c5cae8a625925f9e9
Git Branch: heads/refs/tags/v8.4.0-alpha

**Comment by adi-kmt** [comment]
Hey @lance6716 @jebter, is this issue related to [this](https://github.com/pingcap/tidb/blob/fd6d5c1cd0f6de9b2f000b4ba997cc79f79ae26f/pkg/ddl/db_test.go#L263) comment?

**Comment by adi-kmt** [comment]
So is it as simple as testing the condition and current db?

**Comment by lance6716** [comment]
Yes I guess adding a testing here https://github.com/pingcap/tidb/blob/fd6d5c1cd0f6de9b2f000b4ba997cc79f79ae26f/pkg/ddl/create_table.go#L789-L790 is enough. But you need to find the correct way to do the check. Maybe refer to `ConvertIntToInt` function. And the `lowerBound`/`upperBound` should be decided by column type.

**Comment by adi-kmt** [comment]
So can I take this up?

**Comment by takaidohigasi** [comment]
Can I take over this issue? @lance6716

**Comment by dveeden** [comment]
Looks like the error might differ from the error that MySQL uses in similar cases. Maybe this can be fixed as well?

Not that behavior might depend on the SQL mode.

**Comment by takaidohigasi** [comment]
thank you for the feedback for sql_mode.

I missed it but fix the problem taking the sql_mode into account.
